src/main/groovy/life/qbic/portal/infrastructure/services/workflowtracking/WorkflowTrackingService.groovy

```diff
@@ -15,21 +15,30 @@ class WorkflowTrackingService extends Service {
 
     WorkflowTrackingService(URL rootUrl) {
         super(ServiceType.WORKFLOW_TRACKING, rootUrl)
+        init()
+    }
+
+    WorkflowTrackingService(Service service) {
```

Comment:
I suggest to get rid of this class. Mainly because it will produce confusion if we have several Service class extensions with different behaviour, which is not obvious from its name. 

Just create an `Endpoints` class, that holds the paths to the different endpoints as public static member variables, such as `WORKFLOWS`, `WORFLOWS_INFO`, etc, which you can then implement in your code with `Endpoints.WORKFLOWS`.

The complete URL shall remain dynamic, so you can keep doing: `new URL(service.rootUrl.toExternalForm() + Endpoints.WORKFLOWS)` later, when you make the call to the service.

Reply:
Will do that. 
The service class does not behave differently it only provides additional fields. One of the original reasons for implementing a helper class was that the Service class implements it's endpoints as field as well (health and routes endpoints). 
In addition to that I think that the ability to call `myservice.getWorkflowITracesEndpoint(runId)` improves readability compared to `new URL(myservice.getRootUrl() + Endpoints.WORKFLOW_TRACES + runId)`